### ISSUE TITLE
Rule Proposal: `vue/enforce-style-attribute`

### ISSUE THREAD

**Issue by mussinbenarbia** [vuejs/eslint-plugin-vue#2109]
**Please describe what the rule should do:**

Many projects agree on using one specific way of writing CSS. For example, using CSS Modules, strictly using Scoped CSS, or a combination of the two, which may be rare but could happen when migrating from one style to another. 

This rule helps developers enforce one or either of the two attributes in SFCs.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

#### Option `either`: No warning as long as one attribute is specified. Does not enforce one specific style.
```vue
<!-- ✓ GOOD -->
<style scoped></style>
<style lang="scss" src="../path/to/style.scss" scoped></style>

<!-- ✓ GOOD -->
<style module></style>

<!-- ✗ BAD -->
<style></style>
```

#### Option `scoped`: Style tags must have the `scoped` attribute.
```vue
<!-- ✓ GOOD -->
<style scoped></style>
<style lang="scss" src="../path/to/style.scss" scoped></style>

<!-- ✗ BAD -->
<style module></style>
<style></style>
```

#### Option `module`: Style tags must have the `module` attribute.
```vue
<!-- ✓ GOOD -->
<style module></style>


<!-- ✗ BAD -->
<style scoped></style>
<style lang="scss" src="../path/to/style.scss" scoped></style>
<style></style>
```

**Comment by ota-meshi** [comment]
Thank you for the rule proposal.
Sounds good to me.
However, I think your proposed rule is a little too strict.

Also, I have a similar rule available in another plugin.
https://future-architect.github.io/eslint-plugin-vue-scoped-css/rules/enforce-style-type.html
`eslint-plugin-vue-scoped-css`'s rule can use arrays to allow multiple attribute types in your application. You can also force it to be `plain`.

I think it would be better if we changed your implementation rule to accept options similar to the `eslint-plugin-vue-scoped-css`'s rule.
What do you think?

**Comment by mussinbenarbia** [comment]
Thank you for the feedback and for linking to the other plugin! I think that makes sense. I will modify the implementation to accept the same options.
